src/ordered.rs

```diff
@@ -58,7 +58,12 @@ pub enum Method {
     Measure(Duration),
 }
 
+// Inlining this function will cause a compiler error when compiling for target wasm32-unknown-unknown.
+// The compiler thinks that the variable `n_runs` is a constant. It tries to optimize the variable, removing it in the process.
+// When the compiler remove the n_runs variable, it results in an infinite loop.
+// Therefore, it is a compiler bug, not the code.
 #[cfg(feature = "std")]
+#[inline(never)]
```

Comment:
I would gate this 

with `#[cfg_attr(target_family = "wasm", inline(never)]`

I'm curious about how you determined the compiler was miscompiling/generating invalid wasm

Reply:
Good suggestion. I will look into implementing it.

The way I found out was surprisingly silly. I was doing console::log to print every step of the process when I attempted to deserialize a server key. I found out that it was running the same multiply function infinitely. Eventually I made the code log the value of that variable and the code then worked. I was really confused. Why logging variables suddenly make the bug go away?

The reason is that, during compile time, the compiler thinks that variable is redundant. But, when it has to log the value, it is not redundant any more. The code to interact with the Javascript API break the "optimization" by the compiler. Forcing the function to never be inline stops the optimization from happening as well.

Reply:
yep ok understood :)